apps/studio/tests/e2e/singpass.test.ts

```diff
@@ -0,0 +1,171 @@
+import crypto from "crypto"
+import { test as base, expect } from "@playwright/test"
+
+import { env } from "~/env.mjs"
+import { db } from "~/server/modules/database"
+import { LoginPage } from "./fixtures/login"
+
+base.beforeEach(async () => {
+  await db
+    .updateTable("User")
+    .set({
+      name: "",
+      phone: "",
+      singpassUuid: null,
+    })
+    .where((eb) =>
+      eb("name", "!=", "")
+        .or("phone", "!=", "")
+        .or("singpassUuid", "is not", null),
+    )
+    .execute()
+})
+
+interface LoginPageFixture {
+  loginPage: LoginPage
+}
+const test = base.extend<LoginPageFixture>({
+  loginPage: async ({ page }, fixture) => {
+    const loginPage = new LoginPage(page)
+    await fixture(loginPage)
+  },
+})
+
+test("first login with singpass should succeed", async ({
+  page,
+  loginPage,
+}) => {
+  // Arrange
+  const editorEmail = "editor@open.gov.sg"
+  await page.goto("/sign-in")
+  const signinButton = page.getByRole("button", { name: "Sign in" })
+
+  // Act
+  await loginPage.fillEmail(editorEmail)
+  await expect(page.getByText("Enter OTP")).toBeVisible()
+
+  // NOTE: The function for verification of otp does a comparison between the hash of the submitted token
+  // and the VerificationToken.token in db.
+  await loginPage.fillToken(editorEmail)
+  await signinButton.click()
+
+  await loginPage.defaultMockpassLogin()
+  const continueButton = page.getByRole("link", {
+    name: "Continue to Isomer Studio",
+  })
```

Comment:
issue: I think this should be the one that we are asserting, to ensure that users see the "Continue to Isomer Studio" page if it is the first time they log in.

Reply:
heya, could you explain what you mean here

Reply:
Yup, I think the main distinction between the first-time login and subsequent login user is that they are presented with a screen with the big blue button that says "Continue to Isomer Studio", which I think we should be doing the assertion here.

However, we are asserting that the user sees the edit profile details modal, which is not really true to the intent of this test because the user can be an existing user before we toggled on the Singpass login, or the user might be a returning user that has closed the edit profile modal.

Reply:
ok, thanks for this explanation - so to doublecheck my understanding, this button only shows up on the first login? on subsequent logins, they just get shown all the sitse they have access to right?

Reply:
Yup that's correct, and the edit profile details modal if they never added their details before